src/bdd/bdds.jl

```diff
@@ -0,0 +1,1025 @@
+_bdd_nextid = 1
+_bdd_mutex = Threads.SpinLock()
+_bdd_int_typemax = typemax(Int)
+
+"""A Binary Decision Diagram.
+
+- `index`: the vertex variable (-1 if terminal vertex
+- `low`: low child vertex of BDD (undef if terminal vertex)
+- `high`: high child vertex of BDD (undef if terminal vertex)
+- `value`: terminal boolean value
+- `id`: unique identifier
+"""
+mutable struct Diagram
+  "Root vertex variable index (-1 if terminal vertex)."
+  index::Int
+  "Low child vertex of BDD (undef if terminal vertex)."
+  low::Diagram
+  "High child vertex of BDD (undef if terminal vertex)."
+  high::Diagram
+  "Terminal boolean value."
+  value::Bool
+  "Unique identifier."
+  id::Int
+  "Constructs a terminal."
+  function Diagram(v::Bool)::Diagram
+    α = new()
+    α.index, α.value = -1, v
+    lock(_bdd_mutex)
```

Comment:
Is locks used because its possible to build the structure in parallel?
Is it needed for other caases?

Reply:
It's only for concurrency reasons.

Reply:
Sounds good. Mostly wanted to see where the concurrency happens.